lib/AST/ASTImporter.cpp

```diff
@@ -909,8 +909,28 @@ bool ASTNodeImporter::ImportDeclParts(NamedDecl *D, DeclContext *&DC,
                                       DeclarationName &Name, 
                                       NamedDecl *&ToD,
                                       SourceLocation &Loc) {
+  // Check if RecordDecl is in FunctionDecl parameters to avoid infinite loop.
+  // example: int struct_in_proto(struct data_t{int a;int b;} *d);
+  DeclContext *OrigDC = D->getDeclContext();
+  if(isa<RecordDecl>(D) && OrigDC->isFunctionOrMethod()) {
+    FunctionDecl *FunDecl = dyn_cast<FunctionDecl>(OrigDC);
+    if (FunDecl && FunDecl->hasBody()) {
+      SourceRange RecR = D->getSourceRange();
```

Comment:
Do we need to check the source ranges? Can't we only check whether the RecordDecl is a definition? (This declaration specifically.)

Reply:
Unfortunately no. There is no clean way to tell if the RecordDecl is in the parameter list. We rather go the opposite way. If this RecordDecl's DeclContext is our function but still it is not in the function body, than it cannot come from anywhere else but the parameter list.